warp10/src/main/java/io/warp10/script/ext/http/HTTP.java

```diff
@@ -0,0 +1,278 @@
+  public static final String BODY = "body";
+  public static final String RESPONSE = "response";
+
+  private final Arguments args;
+  private final Arguments output;
+
+  protected Arguments getArguments() {
+    return args;
+  }
+  protected Arguments getOutput() {
+    return output;
+  }
+
+  //
+  // Control
+  //
+
+  private final ReentrantLock stackCountersLock = new ReentrantLock();
+  private final WebAccessController webAccessController;
+
+  public HTTP(String name) {
+    super(name);
+
+    String patternConf = WarpConfig.getProperty(HttpWarpScriptExtension.WARPSCRIPT_HTTP_HOST_PATTERNS);
+
+    // If not defined, use already existing StandaloneWebCallService webAccessController which uses Configuration.WEBCALL_HOST_PATTERNS
+    if (null == patternConf) {
+      webAccessController = StandaloneWebCallService.getWebAccessController();
+    } else {
+      webAccessController = new WebAccessController(patternConf);
+    }
+
+    getDocstring().append("Apply an HTTP method over an url and fetch response.");
+
+    args = new ArgumentsBuilder()
+      .addArgument(String.class, METHOD, "The http method.")
+      .addArgument(String.class, URL, "The URL to send the GET request to. Must begin with http:// or https://.")
+      .addOptionalArgument(Map.class, HEADER, "An optional header.", new HashMap<>())
+      .addOptionalArgument(String.class, BODY, "An optional body", "")
+      .build();
+
+    output = new ArgumentsBuilder()
+      .addArgument(List.class, RESPONSE, "A list containing, for each URL, a 4-element list. These 4-element lists contain, in this order, a LONG status code, a STRING status message or an empty STRING if not available, a MAP of headers and a STRING representing a bytes array encoded as base 64.")
+      .build();
+  }
+
+  @Override
+  public WarpScriptStack apply(Map<String, Object> formattedArgs, WarpScriptStack stack) throws WarpScriptException {
+
+    //
+    // Check capability if set in configuration, or else check stack authentication
+    //
+
+    Properties props = WarpConfig.getProperties();
+
+    if (props.containsKey(HttpWarpScriptExtension.HTTP_CAPABILITY)) {
+      if (stack.getAttribute(WarpScriptStack.CAPABILITIES_ATTR) instanceof Capabilities) {
+        Capabilities capabilities = (Capabilities) stack.getAttribute(WarpScriptStack.CAPABILITIES_ATTR);
+
+        if (!(capabilities.containsKey(props.getProperty(HttpWarpScriptExtension.HTTP_CAPABILITY)))) {
+          throw new WarpScriptException("Capability " + HttpWarpScriptExtension.HTTP_CAPABILITY + " is required by function " + getName());
+        }
+      }
+
+    } else if (!stack.isAuthenticated()) {
+      throw new WarpScriptException(getName() + " requires the stack to be authenticated.");
+    }
+
+    //
+    // Retrieve arguments
+    //
+
+    String method = (String) formattedArgs.get(METHOD);
+    Map<Object, Object> properties = (Map) formattedArgs.get(HEADER);
+    String body = (String) formattedArgs.get(BODY);
```

Comment:
the body parameter could contain a STRING or BYTES to let the user choose the charset used.